src/redturtle/prenotazioni/behaviors/booking_folder/notification_appio/adapters.py

```diff
@@ -0,0 +1,53 @@
+# -*- coding: utf-8 -*-
+import os
+from logging import getLogger
+
+from zope.component import adapter
+from zope.component import getUtility
+from zope.interface import implementer
+
+from redturtle.prenotazioni.content.prenotazione import IPrenotazione
+from redturtle.prenotazioni.interfaces import IBookingNotificationSender
+from redturtle.prenotazioni.interfaces import IBookingNotificatorSupervisorUtility
+from redturtle.prenotazioni.interfaces import IPrenotazioneAPPIoMessage
+from redturtle.prenotazioni.interfaces import IRedturtlePrenotazioniLayer
+from redturtle.prenotazioni.io_tools.api import Api
+
+logger = getLogger(__name__)
+
+
+@implementer(IBookingNotificationSender)
+@adapter(IPrenotazioneAPPIoMessage, IPrenotazione, IRedturtlePrenotazioniLayer)
+class BookingTransitionAPPIoSender:
+    def __init__(self, message_adapter, booking, request) -> None:
+        self.message_adapter = message_adapter
+        self.booking = booking
+        self.request = request
+
+    def send(self):
+        message = self.message_adapter.message
+        subject = self.message_adapter.message
+        booking_type = self.booking.get_booking_type()
+        service_code = getattr(booking_type, "service_code", None)
+        if not service_code:
+            return
+        api_key = os.environ.get(service_code)
+        if not api_key:
+            return
+        if getUtility(IBookingNotificatorSupervisorUtility).is_appio_message_allowed(
+            self.booking
```

Comment:
probably api_key check should be moved inside is_appio_message_allowed because if there isn't a key set, we can't send anything

Reply:
We must have only the business logic in the supervisor. The api key check is the tech check.